GitExtUtils/GitUI/ThreadHelper.cs

```diff
@@ -117,6 +119,21 @@ public static async Task JoinPendingOperationsAsync()
             await _joinableTaskCollection.JoinTillEmptyAsync();
         }
 
+        public static void WaitForPendingOperationsAndDoApplicationEvents()
```

Comment:
❕ This is not the normal way to handle JTF. I would need a clear explanation for why `JoinPendingOperationsAsync` cannot be used before we would want to proceed with this approach.